low-level/files/opt/seagate/sspl/setup/sspl_reset.py

```diff
@@ -0,0 +1,83 @@
+import os
+import shutil
+
+from cortx.sspl.bin.sspl_constants import (PRODUCT_FAMILY,
+ file_store_config_path, DATA_PATH)
+from cortx.utils.process import SimpleProcess
+from cortx.utils.service import Service
+from cortx.utils.conf_store import Conf
+from cortx.sspl.bin.error import SetupError
+
+SSPL_CONFIGURED=f"/var/{PRODUCT_FAMILY}/sspl/sspl-configured"
+
+class HardReset:
+    """Performs SSPL Hard reset.
+    'hard' is used to reset configs and clean log directory """
+
+    name = "hard"
+    user_present=False
+
+    def del_file(self, filename):
+        if os.path.exists(filename):
+            os.remove(filename)
+
+    def process(self):
+        # stop sspl service
+        Service('dbus').process('stop', 'sspl-ll.service')
+
+        # Remove sspl_conf
+        self.del_file(file_store_config_path)
+
+        # Remove sspl-configured file
+        self.del_file(SSPL_CONFIGURED)
+
+        # Remove sspl data
+        shutil.rmtree(DATA_PATH, ignore_errors=True)
+
+        # Remove sspl-ll user if preset
+        CMD = "id -u sspl-ll"
+        output, error, returncode = SimpleProcess(CMD).run()
+        if returncode != 0:
+            raise SetupError(returncode, error, CMD)
+        else:
+            self.user_present=True
+
+        if self.user_present:
+            CMD="/usr/sbin/userdel sspl-ll"
+            output, error, returncode = SimpleProcess(CMD).run()
+            if returncode != 0:
+                raise SetupError(returncode, error, CMD)
```

Comment:
if so, we won't see CMD in error message.
But below one can help.
`raise SetupError(returncode, error + "Command: %s", CMD)`
Please just check.

Reply:
`sspl_setup_consul = "%s/sspl_setup_consul -e %s" % (self._script_dir,
                                                                    self.ENVIRONMENT)
                output, error, returncode = SimpleProcess(sspl_setup_consul).run()
                if returncode != 0:
                    raise SetupError(returncode, error, sspl_setup_consul)`

Reply:
I have referred to the same sspl_post_install.py format to keep the code aligned in all files.
Anything else is required?

Reply:
What I meant is the SetupError is designed in a way that the message should have a placeholder in order to add remaining arguments.
Ref: error.py
`self._desc = message % (args)`

Reply:
Done.